src/github-client-finder/repository-finder.ts

```diff
@@ -0,0 +1,135 @@
+import * as ghCore from "@actions/core";
+import * as semver from "semver";
+import * as github from "@actions/github";
+
+import { ClientDetailOverrides, InstallableClient } from "../util/types";
+import {
+    joinList, getBetterHttpError, assert, getPat,
+} from "../util/utils";
+
+export async function findClientVersionFromGithub(client: InstallableClient, desiredVersionRange: semver.Range):
+    Promise<string> {
+    const availableVersions = await findAvailableVersionFromGithub(client);
+    const clientMatchedVersion = findMatchingVersion(client, availableVersions, desiredVersionRange);
+
+    return clientMatchedVersion;
+}
+
+/**
+ * Find available versions from the github releases
+ * @returns list of the available versions for the provided client
+ */
+export async function findAvailableVersionFromGithub(client: InstallableClient):
+    Promise<string[]> {
+    const githubRepositoryPath = ClientDetailOverrides[client]?.githubRepositoryPath;
+    assert(githubRepositoryPath);
+    const slashIndex = githubRepositoryPath.indexOf("/");
+    const owner = githubRepositoryPath.substring(0, slashIndex);
+    const repo = githubRepositoryPath.substring(slashIndex + 1);
+
+    // using github token to avoid api rate limit hit
+    const octokit = github.getOctokit(getPat());
+    let releaseListresponse;
+
+    // API Documentation: https://docs.github.com/en/rest/reference/repos#list-releases
+    // Octokit Documenation: https://octokit.github.io/rest.js/v18#repos-list-releases
+    try {
+        releaseListresponse = await octokit.rest.repos.listReleases({
+            owner,
+            repo,
+        });
+    }
+    catch (err) {
+        throw getBetterHttpError(err);
+    }
+
+    const availableVersions: string[] = releaseListresponse.data.reduce(
+        (versions: string[], versionData: Record<string, string>) => {
+            try {
+                versions.push(versionData.tag_name);
+            }
+            catch (err) {
+                // ignore invalid
+            }
+            return versions;
+        }, new Array<string>()
+    );
+
+    return availableVersions;
+}
+
+function findMatchingVersion(client: InstallableClient, availableVersions: string[], versionRange: semver.Range):
```

Comment:
is this copied verbatim from directory-finder? can you just export it from there?

Reply:
https://github.com/redhat-actions/openshift-tools-installer/pull/25#discussion_r634024649